add more details on about gh-pages and baseURL on hosting-on-github.md (#1346)
gohugoio/hugoDocs: content/en/hosting-and-deployment/hosting-on-github.md

For more advance settings https://github.com/marketplace/actions/hugo-setup 

## Github pages setting

> By default, the github actions pushes the generated content to the `gh-pages` branch. This means github has to serve your `gh-pages` branch instead of the default branch. You can change this setting by going to Settings > Github Pages, and change the source branch to gh-pages

Reviewer: ```suggestion
By default, the GitHub action pushes the generated content to the `gh-pages` branch. This means GitHub has to serve your `gh-pages` branch as a GitHub Pages branch. You can change this setting by going to Settings > GitHub Pages, and change the source branch to `gh-pages`.
```

![Screenshot of the github pages setting for setting source branch](/images/hosting-and-deployment/hosting-on-github/set-correct-branch.jpg)

## Change baseURL in config.toml

> Don't forget to rename your `baseURL` in `config.toml` with the value `https://<USERNAME>.github.io`

Reviewer: ```suggestion
Don't forget to rename your `baseURL` in `config.toml` with the value `https://<USERNAME>.github.io` for your user repository or `https://<USERNAME>.github.io/<REPOSITORY_NAME>` for a project repository.
```
